### PR TITLE
feat: add performance test capability

### PR DESCRIPTION
**Pull request by davecharles**
- Added performance tests using locust.
- Added guidance README to help with deployment into CI or as a scheduled job.
- Updated docker-compose `clamd` image to sensible value.

### CODE REVIEW

File: perf/locustfile.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+    @staticmethod
+    def parse_response(response):
+        """Parse HTTP response data.
+
+        :param (HttpResponse) response: Response to parse.
+        :returns (dict): parsed json data or None if decode failed.
+        """
+        response_data = b''
+        for chunk in response.iter_content(chunk_size=None):
```
Comment (lgarvey) [inline perf/locustfile.py:131]: Minor, but there's probably no benefit to chunking the response if you're just concating it to a var?

Also would `response.json()` not be sufficient?

File: perf/locustfile.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        }
+        """
+        with self.client.post("/v2/scan-chunked",
+                              name="CLEAN",
+                              stream=True,
+                              auth=AUTH,
+                              headers={"Transfer-encoding": "chunked"},
+                              data=valid_file_gen(),  # noqa
+                              catch_response=True) as response:
+            response.success()
```
Comment (lgarvey) [inline perf/locustfile.py:210]: Is that a valid method?

**Review by lgarvey — APPROVED** [review]
Looks good, and the docs are great.

File: perf/locustfile.py (hunk 2 above)
Comment (davecharles) [reply perf/locustfile.py:210]: Yep it's patched by locust and read by the framework when compiling results

File: perf/locustfile.py (hunk 1 above)
Comment (davecharles) [reply perf/locustfile.py:131]: Good spot this is unnecessary - have updated (still think it's important to catch what `requests` might raise hence keeping the abstract User class's method).

**Pull request merged**